crates/hotshot/src/traits/networking/libp2p_network.rs

```diff
@@ -271,6 +287,7 @@ impl<TYPES: NodeType> TestableNetworkingImplementation<TYPES>
                             mesh_n: (expected_node_count / 2 + 3),
                         }))
                         .server_mode(true)
+                        .identity(libp2p_keypair)
```

Comment:
You probably know better, but what is this doing? could it be related to the failures in CI?

Reply:
The solution is probably to cache the validation result so we don't re-run it for every `get_record` call